### PR TITLE
Add GCP HSM wallet

### PR DESCRIPTION
**Pull request by gnardini**
### Description

Added a GCP HSM wallet to the SDK.

A lot of the config and repo setup was copied from the AWS HSM code.

### Other changes

N/A

### Tested

There's a unit test and I also tested with a real HSM hosted in Valora's GCP project.

### Related issues

- Fixes #

### Backwards compatibility

N/A

### Documentation

N/A

### CODE REVIEW

File: packages/sdk/wallets/wallet-hsm-gcp/package.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  "scripts": {
+    "build": "tsc -b .",
+    "clean": "tsc -b . --clean",
+    "docs": "typedoc && ts-node ../../utils/scripts/linkdocs.ts wallet-hsm-gcp",
+    "test": "jest --runInBand",
+    "lint": "tslint -c tslint.json --project .",
+    "prepublishOnly": "yarn build"
+  },
+  "dependencies": {
+    "@celo/utils": "1.3.1-dev",
```
Comment (gastonponti) [inline packages/sdk/wallets/wallet-hsm-gcp/package.json]: Master is already on the 2.1.1

File: packages/sdk/wallets/wallet-hsm-gcp/package.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    "@celo/wallet-base": "1.3.1-dev",
+    "@celo/wallet-remote": "1.3.1-dev",
+    "@celo/wallet-hsm": "1.3.1-dev",
```
Comment (aaronmgdr) [inline packages/sdk/wallets/wallet-hsm-gcp/package.json]: update these to the versions on master

File: packages/sdk/wallets/wallet-hsm-gcp/package.json [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    "@types/debug": "^4.1.5",
+    "@types/secp256k1": "^4.0.0",
+    "eth-lib": "^0.2.8",
+    "ethereumjs-util": "^5.2.0",
+    "bignumber.js": "^9.0.0",
+    "debug": "^4.1.1",
+    "secp256k1": "^4.0.0"
+  },
+  "devDependencies": {
+    "@celo/connect": "1.3.1-dev",
```
Comment (aaronmgdr) [inline packages/sdk/wallets/wallet-hsm-gcp/package.json]: ditto

File: packages/sdk/wallets/wallet-hsm-gcp/package.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
+{
+  "name": "@celo/wallet-hsm-gcp",
+  "version": "1.3.1-dev",
```
Comment (aaronmgdr) [inline packages/sdk/wallets/wallet-hsm-gcp/package.json]: have this match whatever the other wallets are on

**Review by aaronmgdr — COMMENTED** [review]
assuming this works it looks good except for the few specific dependency comments

File: packages/sdk/wallets/wallet-hsm-gcp/src/gcp-hsm-wallet.ts [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+    }
+    return addressToSigner
+  }
+
+  private async getPublicKeyFromVersionName(versionName: string): Promise<BigNumber> {
+    if (!this.client) {
+      throw new Error('GcpHsmWallet needs to be initialised first')
+    }
+    const [pk] = await this.client.getPublicKey({ name: versionName })
+    const derEncodedPk = pk.pem?.split('\n').slice(1, -2).join('').trim()
```
Comment (gastonponti) [inline packages/sdk/wallets/wallet-hsm-gcp/src/gcp-hsm-wallet.ts]: Can you extract this to a different method that explain why are you doing this? (an example of a valid response, or something like it)

File: packages/sdk/wallets/wallet-hsm-gcp/src/gcp-hsm-wallet.ts [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    // @ts-ignore
+    const pubKey = publicKeyFromAsn1(Buffer.from(derEncodedPk, 'base64'))
```
Comment (gastonponti) [inline packages/sdk/wallets/wallet-hsm-gcp/src/gcp-hsm-wallet.ts:54]: What are you ignoring? Casting `derEncondedPk` was not an option?

**Review by gastonponti — COMMENTED** [review]
Update it agains master, change the versions to the latest and fix the nits.
Did you run the tests with the USING_MOCK flag in false?

File: packages/sdk/wallets/wallet-hsm-gcp/src/gcp-hsm-wallet.ts (hunk 6 above)
Comment (alecps) [reply packages/sdk/wallets/wallet-hsm-gcp/src/gcp-hsm-wallet.ts:54]: It looks like pk.pem can be null or undefined. I addressed this by adding an if-statement to throw if this is the case

File: packages/sdk/wallets/wallet-hsm-gcp/src/gcp-hsm-wallet.ts (hunk 5 above)
Comment (alecps) [reply packages/sdk/wallets/wallet-hsm-gcp/src/gcp-hsm-wallet.ts]: Moved this to a helper and added this in the docstring 

"DER (Distinguished Encoding Rules) is a binary encoding for X.509 certificates and private keys. 
 Unlike PEM, DER-encoded files do not contain plain text statements such as -----BEGIN CERTIFICATE-----

 https://www.ssl.com/guide/pem-der-crt-and-cer-x-509-encodings-and-conversions/#:~:text=DER%20(Distinguished%20Encoding%20Rules)%20is,commonly%20seen%20in%20Java%20contexts."

**Review by alecps — APPROVED** [review]
Getting this over the line per Valora's request

**Pull request merged**